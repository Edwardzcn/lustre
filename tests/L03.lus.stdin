1
True
2
False
3
True
4
False
5
False
6
True
