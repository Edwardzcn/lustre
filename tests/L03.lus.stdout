PhaseToCore
===========

node (x.x : int, x.y : bool)
  returns (y)
  assumes ()
  shows ()
let
  y : int
    = ITE(x.y, x.x, 0)
tel

--- Step 1 ---
x.x : int  = 1
x.y : bool  = True
y = 1
--- Step 2 ---
x.x : int  = 2
x.y : bool  = False
y = 0
--- Step 3 ---
x.x : int  = 3
x.y : bool  = True
y = 3
--- Step 4 ---
x.x : int  = 4
x.y : bool  = False
y = 0
--- Step 5 ---
x.x : int  = 5
x.y : bool  = False
y = 0
--- Step 6 ---
x.x : int  = 6
x.y : bool  = True
y = 6
--- Step 7 ---
x.x : int  = (EOF)
